### PR TITLE
 Make more fields respect the bulk_* field paramters. Fixes #1686. 

### LINKED ISSUE #1686: Many vector fields don't take into account bulk field parameters
Currently there is support for bulk field parameters that allow vector fields to be calculated with respect to a different zero point than the natural one in the on-disk data. This is often useful when dealing with velocities as this corresponds to a galilean transformation, but there's some support in yt as well for a bulk magnetic field as well.

Currently the `bulk_velocity` and `bulk_magnetic_field` field parameters are used mostly in the curvilinear vector component fields and in `particle_fields.py`, but are *not* used in the following places:

- [x] `create_magnitude_field` in `vector_operations.py`
- [x] `_divergence` in `vector_operations.py`
- [x] The derived fields in`fluid_fields.py`
- [x] The derived fields in`fluid_vector_fields.py`
- [x] The derived fields in `angular_momentum.py`
- [ ] The derived fields in `magnetic_field.py`

So that the velocity fields and magnetic fields are all calculated with respect to the same zero point, I think we need to systematically go through these fields and make sure they account for the `bulk_velocity` and `bulk_magnetic_field` field parameters.

### PR DESCRIPTION
**Pull request by ngoldbaum**
## PR Summary

Currently many build-in yt derived fields do not respect the `bulk_velocity` or `bulk_magnetic_field` field parameters. This can lead to inconsistencies, especially when combining fields that *do* take into account these field parameters with fields that don't.

This PR systematically goes through the fields that work directly with the components of the cartesian velocity and magnetic field vectors and ensures that the fields will always offset these vectors by `bulk_velocity` or `bulk_magnetic_field`, where appropriate.

Fields that were updated:

* The `_magnitude` field function (so `velocity_magnitude` and `magnetic_field_magnitude`).
* The `_divergence` field function
* The (specific) angular momentum and derived field depending on the angular momentum
* Kinetic Energy
* Vorticity and shear fields
* Magnetic field strength (and therefore the magnetic energy and alfven speed)
* Poloidal and tangential magnetic field

While working on this, I discovered that `obtain_rvec` and `obtain_rv_vec` was duplicated in `geometry_utils.pyx` and `misc_utilities.pyx` and both versions are used in different places in the field machinery. To clean this up, I deleted the version in `geometry_utils.pyx` in favor of the version of `misc_utilities.pyx` which was patched a few years ago to handle velocity and magnetic fields generically. I also deleted a number of other unused cython functions in these two files.

Finally, I also added a wrapper around `np.stack` that handles units properly.

Apologies for the big diff! I couldn't think of a way to break this up into more manageable chunks.

## PR Checklist

- [x] Code passes flake8 checker
- [x] New features are documented, with docstrings and narrative docs
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: yt/fields/angular_momentum.py [hunk 1]
```diff
@@ -26,78 +26,84 @@
 from .vector_operations import \
     create_magnitude_field
 
-from yt.utilities.lib.geometry_utils import \
+from yt.utilities.lib.misc_utilities import \
```
Comment (brittonsmith) [inline yt/fields/angular_momentum.py]: This isn't necessarily the best line to leave this comment, but can `obtain_rvec` and `obtain_rv_vec` be given more descriptive names?  I had in my head the whole time that `obtain_rvec` signified "radial vector" and subsequently that `obtain_rv_rec` meant "radial velocity", but it's relative velocity.  I could see how others might also be confused by this.

Comment (ngoldbaum) [reply yt/fields/angular_momentum.py]: How about `obtain_position_vector` and `obtain_relative_velocity_vector`?

Comment (brittonsmith) [reply yt/fields/angular_momentum.py]: Yeah, I'm totally fine with those.

**Comment by ngoldbaum** [comment]
I've implemented your suggestion to use relative fields. One wrinkle was that we already had particle fields for this purpose but with a somewhat different name, e.g. `'particle_velocity_relative_x'` instead of `'relative_particle_velocity_x'`. Since the latter name is a little bit easier to work with programatically, I think we should deprecate `'particle_velocity_relative'` and `'particle_position_relative'` and components of those fields in favor of `'relative_particle_position'` and `'relative_particle_velocity'`. Does that sound OK?

I'm not sure offhand if this is still passing all the tests but I'll push fixups if there are test failures until everything is copacetic again.

**Comment by brittonsmith** [comment]
Ah, good point about the particle velocity fields.  I am in favor of your suggestions of deprecating `particle_position_relative` and the like.

**Comment by ngoldbaum** [comment]
It looks like we don't have a great way to deprecate aliases right now and we already have many deprecated fields that don't raise warnings so for now I'm just going to comment that they're deprecated. I'll also fix the test failures. Handling magnetic fields properly is tricky!

When this is merged, I'm going to come back and fix make sure these fields raise deprecation warnings. I'd like to avoid explicitly raising a deprecation warning in a field definition since it's the field name that's an issue, not the field function. My current idea is to add a `deprecated` keyword argument to `add_field` and then set that to True for all the existing deprecated fields.

**Comment by brittonsmith** [comment]
I support your ideas re: field deprecation.  This looks great to me.  Nice work!

**Comment by ngoldbaum** [comment]
This should be good to review now. While debugging the test failures I discovered an issue in the Athena frontend that I've gone ahead and fixed. Unfortunately this means the athena answers tests for the temperature field are incorrect. It would be awesome if @jzuhone could give this a look for that reason.

**Pull request merged**